Source/SwiftLintFramework/Rules/ProhibitedSuperRule.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  ProhibitedSuperRule.swift
+//  SwiftLint
+//
+//  Created by Aaron McTavish on 12/12/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct ProhibitedSuperRule: ConfigurationProviderRule, ASTRule, OptInRule {
+    public var configuration = ProhibitedSuperConfiguration()
+
+    public init() { }
+
+    public static let description = RuleDescription(
+        identifier: "prohibited_super_call",
+        name: "Prohibited calls to super",
+        description: "Some methods should not call super",
+        nonTriggeringExamples: [
+            "class VC: UIViewController {\n" +
+                "\toverride func loadView() {\n" +
+                "\t}\n" +
+            "}\n",
+            "class NSView {\n" +
+                "\tfunc updateLayer() {\n" +
+                    "\t\tself.method1()" +
+                "\t}\n" +
+            "}\n"
+        ],
+        triggeringExamples: [
+            "class VC: UIViewController {\n" +
+                "\toverride func loadView() ↓{\n" +
+                    "\t\tsuper.loadView()\n" +
+                "\t}\n" +
+            "}\n",
+            "class VC: NSFileProviderExtension {\n" +
+                "\toverride func providePlaceholder(at url: URL," +
+                "completionHandler: @escaping (Error?) -> Void) ↓{\n" +
+                    "\t\tself.method1()\n" +
+                    "\t\tsuper.providePlaceholder(at:url, completionHandler: completionHandler)\n" +
+                "\t}\n" +
+            "}\n",
+            "class VC: NSView {\n" +
+                "\toverride func updateLayer() ↓{\n" +
+                    "\t\tself.method1()\n" +
+                    "\t\tsuper.updateLayer()\n" +
+                    "\t\tself.method2()\n" +
+                "\t}\n" +
+            "}\n"
+        ]
+    )
+
+    public func validateFile(_ file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard let offset = dictionary["key.bodyoffset"] as? Int64,
+            let name = dictionary["key.name"] as? String
+            else { return [] }
+
+        let substructure = (dictionary["key.substructure"] as? [SourceKitRepresentable]) ?? []
```

Comment:
could move this variable declaration down, and merge both `guard` groups together